pkg/server/auth.go

```diff
@@ -79,16 +98,72 @@ func (s *Server) authRequest(ctx context.Context, csrPEM []byte) (string, bool)
 		return identities, false
 	}
 
-	// ensure identity matches requests URIs
-	if !identitiesMatch(caller.Identities, csr.URIs) {
-		log.Error(fmt.Errorf("%v != %v", caller.Identities, csr.URIs), "failed to match URIs with identities")
+	if s.impersonatedIdentity == "" {
+		if !identitiesMatch(caller.Identities, csr.URIs) {
+			log.Error(fmt.Errorf("%v != %v", caller.Identities, csr.URIs), "failed to match URIs with identities")
+			return identities, false
+		}
+	} else if !identitiesMatch([]string{s.impersonatedIdentity}, csr.URIs) {
+		log.Error(fmt.Errorf("%v != %v", s.impersonatedIdentity, csr.URIs), "failed to match URIs with identities")
```

Comment:
```suggestion
		log.Error(fmt.Errorf("%v != %v", s.impersonatedIdentity, csr.URIs), "failed to match URIs with impersonated identities")
```

nit: might be nice to help distinguish